app/livechat/server/api/v1/session.js

```diff
@@ -0,0 +1,131 @@
+import { check, Match } from 'meteor/check';
+
+import { API } from '../../../../api';
+import { Livechat } from '../../lib/Livechat';
+import { LivechatSessions } from '../../../../models';
+
+API.v1.addRoute('livechat/session.incVisitCount/:token', {
+	post() {
+		try {
+			check(this.urlParams, {
+				token: String,
+			});
+
+			const { count } = LivechatSessions.updateSessionCount(this.urlParams.token);
+			return API.v1.success({ count });
+		} catch (e) {
+			return API.v1.failure(e);
+		}
+	},
+});
+
+API.v1.addRoute('livechat/session.visitorInfo/:token', {
+	get() {
+		try {
+			check(this.urlParams, {
+				token: String,
+			});
+
+			const session = LivechatSessions.findOneByToken(this.urlParams.token);
+			return API.v1.success(session);
+		} catch (e) {
+			return API.v1.failure(e);
+		}
+	},
+});
+
+API.v1.addRoute('livechat/session.register', {
+	post() {
+		try {
+			check(this.bodyParams, {
+				token: String,
+				location: Match.Maybe(Match.ObjectIncluding({
+					city: String,
+					countryCode: String,
+					countryName: String,
+					latitude: Number,
+					longitude: Number,
+				})),
+				deviceInfo: Match.ObjectIncluding({
+					os: String,
+					osVersion: Number,
+					browserName: String,
+					browserVersion: Number,
+				}),
+			});
+
+			const { token, location, deviceInfo } = this.bodyParams;
+			const existingSession = LivechatSessions.findOne({ token });
```

Comment:
Let's use the new method you added  [here](https://github.com/RocketChat/Rocket.Chat/pull/14813/files#diff-60705f83eda16683bd76861a949764d3R33) to find the session.